packages/yourloops/test/integration/assert/header.ts

```diff
@@ -25,34 +25,91 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-import { type BoundFunctions, type queries, screen, within } from '@testing-library/react'
+import { type BoundFunctions, fireEvent, type queries, screen, within } from '@testing-library/react'
+import { UserRole } from '../../../lib/auth/models/enums/user-role.enum'
+import { type Team } from '../../../lib/team'
+import userEvent from '@testing-library/user-event'
 
-const checkHeader = (header: BoundFunctions<typeof queries>, fullName: string) => {
+const checkHeader = (header: BoundFunctions<typeof queries>) => {
   expect(header.getByLabelText('YourLoops Logo')).toBeVisible()
   expect(header.getByLabelText('Go to notifications list')).toBeVisible()
-  expect(header.getByText(fullName)).toBeVisible()
 }
 
-export const checkHCPHeader = (fullName: string) => {
+const getIconTestIdByRole = (role: UserRole): string => {
+  switch (role) {
+    case UserRole.Caregiver:
+      return 'caregiver-icon'
+    case UserRole.Hcp:
+      return 'hcp-icon'
+    case UserRole.Patient:
+      return 'patient-icon'
+  }
+}
+
+const checkUserMenu = async (header: BoundFunctions<typeof queries>, userName: string, role: UserRole) => {
+  const iconTestId = getIconTestIdByRole(role)
+
+  expect(header.getByTestId(iconTestId)).toBeVisible()
+  expect(header.getByText(userName)).toBeVisible()
+
+  fireEvent.click(header.getByText(userName))
+
+  const userMenu = within(screen.getByTestId('user-menu'))
+  expect(userMenu.getByText('Profile settings')).toBeVisible()
+  expect(userMenu.getByText('Customer support')).toBeVisible()
+  expect(userMenu.getByText('Logout')).toBeVisible()
+
+  await userEvent.click(screen.getByRole('presentation').firstChild as HTMLElement)
+  expect(screen.queryByTestId('user-menu')).not.toBeInTheDocument()
+}
+
+const checkTeamScopeMenu = async (header: BoundFunctions<typeof queries>, selectedTeamParams: { teamName: string, isPrivate?: boolean }, availableTeams: Team[]) => {
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review (if it exists :^)  -->
**👏 praise:** ‏Really good